### ISSUE TITLE
API rate limiting

### ISSUE THREAD

**Issue by Akkuma** [rstacruz/pnpm#536]
### pnpm version:
v0.48.0

### Code to reproduce the issue:
`pnpm i` delete pnpm cache/store several times rinse and repeat or `pnpm update` several times

### Expected behavior: 
Can `pnpm i` without running into rate limiting issues

### Actual behavior:
pnpm gets rate limited

### Additional information:
```json
{
  "message": "API rate limit exceeded for [IP ADDRESS REMOVED]. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
  "documentation_url": "https://developer.github.com/v3/#rate-limiting"
}
```
 - `node -v` prints:
 - Windows, OS X, or Linux?: OS X

**Comment by Akkuma** [comment]
To be fair, this is a lot harder to hit than you'd think, but I took off 10 api hits the first time around and 6 from a pnpm update. If you start trashing node_modules and the like quite a bit you could run into this like I did. I think an acceptable short term solution would be to simply inform the user of what is actually going on here. The long term solution is probably needed if you want to use pnpm on a build server, which might hit this quite quickly.

**Comment by zkochan** [comment]
I know about this issue. TBH it happens quite often in the pnpm tests. And I even reverted the Git API usage a few releases ago. However, with the new global store, I have to know the commit sha in order to create a unique id for the dependency. Using GitHub's API was the easiest way to achieve this. In long term we'll have to rewrite it to work without the API usage

**Comment by iamstarkov** [comment]
@zkochan how does npm do it?

**Comment by zkochan** [comment]
Using the git CLI. Shouldn't be hard to implement this, it is just that we currently use `codeload.github.com` to download the tarball instead of using git commands. Using GitHub API + codeload is probably faster than using Git

**Comment by zkochan** [comment]
I'll try to do this tonight using [this feature](https://git-scm.com/docs/git-ls-remote) of git to resolve tags/branches + cloning a single commit

**Comment by zkochan** [comment]
Published as not latest for now to version v0.49.1. You can install pnpm@next. I wan to review the solution today because it seems a lot slower.

**Comment by zkochan** [comment]
Published a faster fix to [v0.49.2](https://github.com/rstacruz/pnpm/releases/tag/v0.49.2). With the `next` dist-tag for now
